verify/helpers_test.go

```diff
@@ -0,0 +1,200 @@
+			profileStatus: "BLOCKED",
+			chaincode:     "",
+			expectedErr:   errors.New("chaincode is blocked"),
+		},
+		{
+			name:          "invalid chaincode",
+			userId:        "4",
+			profileURL:    "http://test.com",
+			profileStatus: "BLOCKED",
+			chaincode:     123,
+			expectedErr:   errors.New("chaincode is not a string"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			client.Collection("users").Doc(testCase.userId).Set(ctx, map[string]interface{}{
+				"chaincode":     testCase.chaincode,
+				"profileURL":    testCase.profileURL,
+				"profileStatus": testCase.profileStatus,
+			})
+
+			profileURL, profileStatus, chaincode, err := getUserData(client, ctx, testCase.userId)
+
+			assert.Equal(t, testCase.expectedErr, err)
+			if testCase.expectedErr == nil {
+				assert.Equal(t, testCase.profileURL.(string), profileURL)
+				assert.Equal(t, testCase.profileStatus.(string), profileStatus)
+				assert.Equal(t, testCase.chaincode.(string), chaincode)
+			}
+		})
+	}
+}
+
+func TestGetUserIdFromBody(t *testing.T) {
+	// valid request body
+	body := []byte(`{"userId": "123"}`)
+	expected := "123"
+
+	actual := getUserIdFromBody(body)
+
+	if actual != expected {
+		t.Errorf("getUserIdFromBody returned %v, expected %v", actual, expected)
+	}
+
+	// empty request body
+	body = []byte(``)
+	expected = ""
+
+	actual = getUserIdFromBody(body)
+
+	if actual != expected {
+		t.Errorf("getUserIdFromBody returned %v, expected %v", actual, expected)
+	}
+
+	// invalid request body
+	body = []byte(`{"invalidProperty": ""}`)
+	expected = ""
+
+	actual = getUserIdFromBody(body)
+
+	if actual != expected {
+		t.Errorf("getUserIdFromBody returned %v, expected %v", actual, expected)
+	}
+}
+
+type testVerifyData struct {
+	name           string
+	chaincode      string
+	mockStatusCode int
+	mockResBody    string
+	expectedStatus string
+	expectedErr    error
+}
+
+func TestVerify(t *testing.T) {
+	testCases := []testVerifyData{
+		{
+			name:           "VERIFIED",
+			chaincode:      "testchaincode",
+			mockStatusCode: http.StatusOK,
+			mockResBody:    `{"hash": "$2a$12$ScGc2Q0t0rqqSJK1E2W/WuaRVAchaVWdUqb1hQi21cFTnOVvlIdry"}`,
+			expectedStatus: "VERIFIED",
+			expectedErr:    nil,
+		},
+		{
+			name:           "BLOCKED",
+			chaincode:      "invalid",
+			mockStatusCode: http.StatusForbidden,
+			mockResBody:    `{"hash": "abcdefghijklmnopqrstuvwxyz"}`,
+			expectedStatus: "BLOCKED",
+			expectedErr:    nil,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(testCase.mockStatusCode)
+				w.Write([]byte(testCase.mockResBody))
+			}))
+
+			status, err := verify(server.URL+"/", testCase.chaincode)
+
+			assert.Equal(t, testCase.expectedStatus, status)
+			assert.Equal(t, testCase.expectedErr, err)
+
+			defer server.Close()
```

Comment:
Could you check if you can use a `httptest.NewRecorder()` 
it records the response. Then you can directly send the response by making a new Request. making a HTTP request is better than directly calling up the server.

You could take a reference from here: https://github.com/Real-Dev-Squad/website-learn-backend/blob/4c062abaf2e8d3da61fa5a9b99dec2fb5ba45256/src/tests/dashboard_test.go#L15

Reply:
The reason for using `httptest.NewServer()` is that we can mock the response instead of making a request to the profile service. Not sure if we can simulate the same using `httptest.NewRecorder()` though. I'll have to look into this